src/paths/dijkstra.c

```diff
@@ -1038,25 +1112,50 @@ int igraph_get_all_shortest_paths_dijkstra(const igraph_t *graph,
             tmp = igraph_vector_tail(path);
             if (is_target[(long int)tmp] == 1) {
                 /* we need this path, keep it */
-                VECTOR(*res)[j] = path;
+                if (!free_res){
+                    VECTOR(*res)[j] = path;
+                }
+                if (res_e){
+                    path = (igraph_vector_t*)VECTOR(*res_e)[i];
+                    VECTOR(*res_e)[j] = path;
+                }
                 j++;
             } else {
                 /* we don't need this path, free it */
-                igraph_vector_destroy(path); free(path);
+                if (!free_res){
+                    igraph_vector_destroy(path); free(path);
+                }
+                if (res_e){
+                    path = (igraph_vector_t*)VECTOR(*res_e)[i];
+                    igraph_vector_destroy(path); free(path);
+                }
             }
         }
-        IGRAPH_CHECK(igraph_vector_ptr_resize(res, j));
-
-        /* sort the paths by the target vertices */
-        igraph_vector_ptr_sort(res, igraph_i_vector_tail_cmp);
+        if (!free_res){
+            IGRAPH_CHECK(igraph_vector_ptr_resize(res, j));
+            /* sort the paths by the target vertices */
+            igraph_vector_ptr_sort(res, igraph_i_vector_tail_cmp);
+        }
+        
+        if (res_e){
+            IGRAPH_CHECK(igraph_vector_ptr_resize(res_e, j));
+            igraph_vector_ptr_sort(res_e, igraph_i_vector_tail_cmp);
+        }
     }
 
     /* free the allocated memory */
     igraph_vector_destroy(&order);
     igraph_Free(is_target);
     igraph_vector_destroy(&dists);
     igraph_vector_ptr_destroy_all(&parents);
-    IGRAPH_FINALLY_CLEAN(4);
-
+    igraph_vector_ptr_destroy_all(&parents_edge);
+    if (free_res){
+        igraph_vector_ptr_destroy_all(res);
+        IGRAPH_FINALLY_CLEAN(6);
+        igraph_free(res);
+    }
+    else{
+        IGRAPH_FINALLY_CLEAN(5);
```

Comment:
It's just a matter of taste, but I'd rather have an unconditional `IGRAPH_FINALLY_CLEAN(5)` outside the `if` condition, and an extra `IGRAPH_FINALLY_CLEAN(1)` in the `if`.

Reply:
I will shuffle the ordering around, both here and where the `vertices` vector is allocated in case the user did not provide one.